crates/holochain/src/conductor/conductor.rs

```diff
@@ -2737,12 +2741,24 @@ async fn p2p_event_task(
                     >= NUM_PARALLEL_EVTS)
                     .then(std::time::Instant::now);
 
-                if let Err(e) = handle.dispatch_holochain_p2p_event(evt).await {
-                    tracing::error!(
-                        message = "error dispatching network event",
-                        error = ?e,
-                    );
+                // This loop is critical, ensure that nothing in the dispatch kills it by blocking permantantly
+                tokio::select! {
```

Comment:
To be doubly sure that this loop does not block, limit its runtime. It shouldn't really need more than 30s and if it does, there's a decent chance something else needs to do something more than this needs to keep waiting

Reply:
[tokio::time::timeout](https://docs.rs/tokio/1.28.2/tokio/time/fn.timeout.html) may be more ergonomic here.

Reply:
Yes it would be, I'll look at updating this while I'm waiting on my environment to be ready